app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -213,20 +215,55 @@ export default Component.extend(
       });
     },
 
+    _destroyTippyInstances() {
+      this.tippyInstances.forEach((instance) => {
+        instance.destroy();
+      });
+      this.tippyInstances = [];
+    },
+
     @on("didInsertElement")
     _composerEditorInit() {
       const $input = $(this.element.querySelector(".d-editor-input"));
 
       if (this.siteSettings.enable_mentions) {
         $input.autocomplete({
           template: findRawTemplate("user-selector-autocomplete"),
-          dataSource: (term) =>
-            userSearch({
+          dataSource: (term) => {
+            this._destroyTippyInstances();
+            return userSearch({
               term,
               topicId: this.topic?.id,
               categoryId: this.topic?.category_id || this.composer?.categoryId,
               includeGroups: true,
-            }),
+            }).then((result) => {
+              if (result?.users?.length > 0) {
+                result.users.forEach((user, index) => {
```

Comment:
I generally do something like `(result.users || [])`, it's more elegant than the if above IMO.

If you want to keep the `if` do it as a guard:

```
if (result?.users?.length < 1) {
  return;
}
```

Reply:
Much cleaner. I agree!